### PR TITLE
Async await support

### PR DESCRIPTION
**Pull request by delvedor**
As titled! :)

Currently the *async/await* is supported only when using `.add`, `.on('request', fn)` is not supported because is an EventEmitter. Do you know if there is any workaround to support this as well?

### CODE REVIEW

File: upring.js [hunk 1]
```diff
@@ -51,7 +51,12 @@ function UpRing (opts) {
     if (this._router) {
       func = this._router.lookup(req)
       if (func) {
-        func(req, reply)
+        var result = func(req, reply)
+        if (result && typeof result.then === 'function') {
+          result
+            .then(res => reply(null, res))
```
Comment (mcollina) [inline upring.js]: can you call reply using nextTick? Otherwise if that throws it will be swallowed by the promise.

File: upring.js [hunk 2]
```diff
@@ -51,7 +51,12 @@ function UpRing (opts) {
...
+            .catch(err => reply(err, null))
```
Comment (mcollina) [inline upring.js]: same for catch

**Comment by mcollina** [comment]
Can you please add it to the docs? Also, you'd need to add node 8 to .travis.yml, or the test would not run.

You can use https://github.com/ljharb/util.promisify if you want to add an API for `request` as well, like `requestp`.

Can you also add a test with plain promises so that we can run that in Node < 8?

File: upring.js [hunk 3]
```diff
@@ -256,7 +257,7 @@ UpRing.prototype.request = function (obj, callback, _count) {
   }
 
   return this
-}
+})
```
Comment (mcollina) [inline upring.js]: Can you call the promisified version requestp, or something different? I do not want request() users to pay the cost of promisified code.

File: upring.js [hunk 4]
```diff
@@ -11,6 +11,7 @@ const networkAddress = require('network-address')
 const bloomrun = require('bloomrun')
 const pino = require('pino')
 const tinysonic = require('tinysonic')
+// const promisify = require('util.promisify')
```
Comment (mcollina) [inline upring.js]: is this a leftover?

File: upring.js [hunk 5]
```diff
@@ -253,6 +259,19 @@ UpRing.prototype.request = function (obj, callback, _count) {
   return this
 }
 
+UpRing.prototype.requestp = function (obj) {
+  return new Promise(requestp.bind(this))
```
Comment (mcollina) [inline upring.js]: this bind is not useful. can we use `util.promisify` instead?

File: upring.js (hunk 4 above)
Comment (delvedor) [inline upring.js]: Yes 😅

File: upring.js (hunk 5 above)
Comment (mcollina) [inline upring.js]: if you want to use `bind`, you can avoid that and do `that = this`.

Comment (delvedor) [inline upring.js]: Do you want something like this?
```js
UpRing.prototype.requestp = function (obj) {
  return promisify(this.request)(obj)
}
```

Comment (mcollina) [inline upring.js]: no.
`UpRing.prototype.requestp = promisify(Upring.prototype.request)` might work.

**Pull request merged**